crates/history/src/listener.rs

```diff
@@ -0,0 +1,9 @@
+use std::rc::Rc;
+
+/// A History Listener to manage callbacks registered on a [`History`].
+///
+/// This Listener has the same behaviour as the [`EventListener`] from [`gloo`]
+/// that the underlying callback will be unregistered when the listener is dropped.
+pub struct HistoryListener {
```

Comment:
This should have the [`must_use` attribute](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute).